controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -1612,6 +1640,12 @@ private boolean checkCoverage(double keyStart, double keyEnd, TreeMap<Double, Do
                 .thenApply(x -> !x);
     }
 
+    private CompletableFuture<Boolean> isScopeDeleted(String scope, OperationContext context) {
```

Comment:
The correct way to make sure that a scope is indeed deleted would be as follows:

```
return !checkScopeInDeletingTable && !checkScopeExists
```

The current implementation assumes this method is invoked after the scope is added to` _deletingScopes` Table.

Reply:
yes, this method is only called once and that too as part of `checkDone`. Removing the scope from `deleting_scopes_table` is the last step of the entire process.